src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticationBuilderExtensions.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.WebAssembly.Authentication;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extensions for remote authentication services.
+    /// </summary>
+    public static class RemoteAuthenticationBuilderExtensions
+    {
+        /// <summary>
+        /// Replaces the existing <see cref="UserFactory{TAccount}"/> with the user factory defined by <typeparamref name="TUserFactory"/>.
+        /// </summary>
+        /// <typeparam name="TRemoteAuthenticationState">The remote authentication state.</typeparam>
+        /// <typeparam name="TAccount">The account type.</typeparam>
+        /// <typeparam name="TUserFactory">The new user factory type.</typeparam>
+        /// <param name="builder">The <see cref="IRemoteAuthenticationBuilder{TRemoteAuthenticationState, TAccount}"/>.</param>
+        /// <returns>The <see cref="IRemoteAuthenticationBuilder{TRemoteAuthenticationState, TAccount}"/>.</returns>
+        public static IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount> AddUserFactory<TRemoteAuthenticationState, TAccount, TUserFactory>(
```

Comment:
Should this be called `SetUserFactory`?

It appears that you can only have one, so `Add` is misleading.

Reply:
Same for the other overloads below of course.

Reply:
I followed the language Identity uses to keep consistency and they call these things 'AddUserManager` and so on. I might be nittpicking, but we haven't used `Set` ever in our APIs in this space. I'm just trying to be consistent with the previous art.

Reply:
OK, it's not something I'm massively troubled by. It does seem a touch odd but then arguably people should understand that one one such factory can be in effect (it's meaningless to chain them), so the "add" terminology isn't going to throw anyone off track too far.